app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomFragment.kt

```diff
@@ -134,9 +174,68 @@ class ChatRoomFragment : Fragment(), ChatRoomView {
     private fun setupComposer() {
         if (isChatRoomReadOnly) {
             text_room_is_read_only.setVisible(true)
-            top_container.setVisible(false)
+            input_container.setVisible(false)
         } else {
-            text_send.setOnClickListener { sendMessage(text_message.textContent) }
+            var playAnimation = true
+            text_message.asObservable(0)
+                .subscribe({ t ->
+                    if (t.isNotEmpty() && playAnimation) {
+                        button_show_attachment_options.fadeInOrOut(1F, 0F, 120)
+                        button_send.fadeInOrOut(0F, 1F, 120)
+                        playAnimation = false
+                    }
+
+                    if (t.isEmpty()) {
+                        button_send.fadeInOrOut(1F, 0F, 120)
+                        button_show_attachment_options.fadeInOrOut(0F, 1F, 120)
+                        playAnimation = true
+                    }
+                })
+
+            button_send.setOnClickListener { sendMessage(text_message.textContent) }
+
+
+            button_show_attachment_options.setOnClickListener {
+                if (layout_message_attachment_options.isShown) {
+                    hideAttachmentOptions()
+                } else {
+                    showAttachmentOptions()
+                }
+            }
+
+            view_dim.setOnClickListener { hideAttachmentOptions() }
+
+            button_files.setOnClickListener {
+                handler.postDelayed({
```

Comment:
`handler.postDelayed()` with an anonymous `Runnable` is a source of a lot of crashes.
If you rotate the screen, or finish the activity before the delay, it will crashe accessing an invalid Activity and invalid views.

Reply:
I did human tests here and get no crash when rotating the screen, finishing the activity, etc. Could you check it too?

Reply:
It is a race condition and it will crash at some point...

When using `postDelayed` you should always call `removeCallbacks(runnable)` on the appropriate lifecycle callback.

Reply:
Thank you @luciofm 👍